spec/reek/smell_detectors/duplicate_method_call_spec.rb

```diff
@@ -126,6 +126,40 @@ def alfa
     end
   end
 
+  context 'with repeated method calls inside blocks' do
+    it 'does not report calls to parameters of different blocks' do
+      src = <<-EOS
+        def alfa(bravo, charlie)
+          bravo.each { |delta| delta.echo }
+          charlie.each { |delta| delta.echo }
+        end
+      EOS
+      expect(src).not_to reek_of(:DuplicateMethodCall)
+    end
+
+    it 'does not report calls inside lambda expressions' do
+      src = <<-EOS
+        def alfa
+          bravo = ->(delta) { delta.echo.foxtrot }
```

Comment:
Why do you need the "foxtrot" at the end? We can drop that, can't we?

Reply:
Probably :-)